File: oop-backend/src/main/java/oop_backend/oop/model/WorldProblem.java
package oop_backend.oop.model;

import java.util.List;

public class WorldProblem {
    private String id;
    private String description;
    private List<ProblemOption> options;
    
    public WorldProblem() {}
    
    public WorldProblem(String id, String description, List<ProblemOption> options) {
        this.id = id;
        this.description = description;
        this.options = options;
    }
    
    // Getters and Setters
    public String getId() {
        return id;
    }
    
    public void setId(String id) {
        this.id = id;
    }
    
    public String getDescription() {
        return description;
    }
    
    public void setDescription(String description) {
        this.description = description;
    }
    
    public List<ProblemOption> getOptions() {
        return options;
    }
    
    public void setOptions(List<ProblemOption> options) {
        this.options = options;
    }
}

File: oop-frontend/src/models/WorldProblem.js
import { ProblemOption } from './ProblemOption';

export class WorldProblem {
    constructor(data = {}) {
        this.description = data.description || '';
        this.options = data.options ? 
            data.options.map(option => new ProblemOption(option)) : 
            [];
    }

    getDescription() {
        return this.description;
    }

    setDescription(description) {
        this.description = description;
    }

    getOptions() {
        return this.options;
    }

    setOptions(options) {
        this.options = options.map(option => 
            option instanceof ProblemOption ? option : new ProblemOption(option)
        );
    }

    static fromApi(apiData) {
        return new WorldProblem(apiData);
    }
}